### PR TITLE
Add `SpglibDataset` and `SpglibMagneticDataset` and  dataclass

### LINKED ISSUE #491: dataclass output for Python API
We start using `dataclasses.dataclass` for some outputs of Python API.

- https://github.com/spglib/spglib/pull/485
- https://github.com/spglib/spglib/pull/490

### PR DESCRIPTION
**Pull request by lan496**
Introduces `dataclasses.dataclass` for `get_symmetry_dataset` and `get_magnetic_symmetry_dataset` in Python API.

### CODE REVIEW

File: docs/dataset.md [hunk 1]
```diff
@@ -244,7 +244,7 @@ is obtained.
 %    groups (Seto's web site)](http://pmsl.planet.sci.kobe-u.ac.jp/~seto/?page_id=37&lang=en).
 
 `pointgroup_symbol` is the symbol of the crystallographic point
-group in the Hermann–Mauguin notation. There are 32 crystallographic
+group in the Hermann-Mauguin notation. There are 32 crystallographic
```
Comment (LecrisUT) [inline docs/dataset.md:247]: Good catch

File: python/spglib/spglib.py [hunk 2]
```diff
@@ -139,10 +139,267 @@ def __iter__(self) -> Iterator[str]:
...
+    rotations: NDArray[np.intc]
+    """Matrix (rotation) parts of space group operations.
+
+    shape=(n_operations, 3, 3), order='C', dtype='intc'
+
+    Space group operations are obtained by
+
+    .. code-block:: python
+
+        [(r,t) for r, t in zip(rotations, translations)]
```
Comment (LecrisUT) [inline python/spglib/spglib.py]: Put this in a `..note::` or higher.

Similar for `translations`

**Review by LecrisUT — APPROVED** [review]
Sorry, I had this pending review, but forgot to hit submit review. Only 1 minor note

File: python/spglib/spglib.py (hunk 2 above)
Comment (lan496) [reply python/spglib/spglib.py]: Done
image

**Review by LecrisUT — APPROVED** [review]

**Comment by lan496** [comment]
@LecrisUT These required test failed somehow. Can you check it?
https://github.com/spglib/spglib/pull/490/checks?check_run_id=25990624567

**Comment by LecrisUT** [comment]
Fedora 40 was released so F38 was retired. This changes every 6 months unfortunately

**Comment by lan496** [comment]
I see. I've updated the branch protection rule.

**Pull request merged**